hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/driver/impl/StateStoreFileBaseImpl.java

```diff
@@ -348,25 +357,28 @@ public <T extends BaseRecord> boolean putAll(
     for (Entry<String, T> entry : toWrite.entrySet()) {
       String recordPath = entry.getKey();
       String recordPathTemp = recordPath + "." + now() + TMP_MARK;
-      BufferedWriter writer = getWriter(recordPathTemp);
+      BufferedWriter writer = getBufferedWriter(recordPathTemp);
+      boolean recordWrittenSuccessfully = true;
```

Comment:
Initialize the value to false and then only set it to true when the write completes.